buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.java

```diff
@@ -54,6 +63,51 @@ public void apply(Project project) {
                 }
               });
     }
+    if (System.getenv().containsKey("METALAVA_BINARY_PATH")) {
```

Comment:
pls extract all this logic in a separate method with a descriptive name, additionally avoid nesting logic in `if`s like you're doing here, instead consider

```
if (!foo) { return; }
// stuff
if (!bar) { return; }
// other stuff.
```